### ISSUE TITLE
Sclang doesn't connect to supernova over TCP

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#4062]
Environment
-----------

* Your SuperCollider version: 3.10dev
* Your operating system and version: macOS 10.13

Steps to reproduce (for bugs)
-----------------------------

```supercollider
Server.supernova;
s.options.protocol_(\tcp);
// s.addr.port_(57117); //if we need to override
(
s.waitForBoot({
	"booted!".postln; //we never get here
	x = {PinkNoise.ar(-12.dbamp)}.play;
});
)
x.free;
s.quit;
```

Error message (for bugs)
------------------------

```
...
NetAddr-Connect failed with exception: connect: Connection refused
NetAddr-Connect failed with exception: connect: Connection refused
WARNING: Couldn't connect to TCP address 127.0.0.1:57110
```

Expected Behavior
-----------------

Supernova boots and is usable.

Current Behavior
----------------

Supernova boots but sclang does not seem to be connected to it.

In the process of debugging it, I discovered that NetAddr fails to connect while Supernova is booting, since it runs out of reconnection attempts. After I changed this in ```Server.sc```, in the class ```Server```,
```SuperCollider
	bootServerApp { |onComplete|
(...)
			if(options.protocol == \tcp, { addr.tryConnectTCP(onComplete, maxAttempts: 30) }, onComplete);
		}
	}
```
it seemed to help, as there was no error. However, the booting process did not seem to finish properly, as the function in the ```.waitForBoot``` Routine never gets executed, and I usually can't play any sounds. On the few attempts I was able to start synths after booting, the server status widget - CPU load, number of synths etc - was not updating, as if the server was non-resposive.

Supernova also seems to not exit cleanly after using TCP, as I can't reuse the same port to boot the server again.

**Comment by miguel-negrao** [comment]
Does increasing the number of attempts [here](https://github.com/miguel-negrao/supercollider/blob/04cfe89b0959a0aac351b9c6e70b9c24e4e44d18/SCClassLibrary/Common/Control/Server.sc#L773) help ?

**Comment by sonoro1234** [comment]
Is it possible to write the tcp strings sended by supernova to sclang? (specifically those badly constructed)

**Comment by dyfer** [comment]
Thanks @sonoro1234!
The issue of not connecting to supernova on TCP indeed seems to be solved by https://github.com/supercollider/supercollider/pull/4435 (which is reformatted #4410). 

The issue of server status panel in the IDE not updating is still valid, but affects both scsynth and supernova and was reported in https://github.com/supercollider/supercollider/issues/756

I'll close this now. Thanks again!

**Comment by brianlheim** [comment]
Please don't close issues until the related PR is merged, thanks. :)

**Comment by dyfer** [comment]
fixed in #4435
